src/handlers.rs

```diff
@@ -261,8 +387,203 @@ fn serialize_initgroups(groups: Vec<Gid>) -> Result<Vec<u8>> {
     Ok(result)
 }
 
+impl Hostent {
+    fn serialize(&self) -> Result<Vec<u8>> {
```

Comment:
To match the rest of the structure of this file, I'd probably change this to be a
```rust
fn serialize_hostent(hostent: Hostent) -> Result<Vec<u8>> {
  // …
}
```